homeassistant/components/qbittorrent/sensor.py

```diff
@@ -15,111 +17,211 @@
+    _attr_has_entity_name = True
 
     def __init__(
         self,
-        description: QBittorrentSensorEntityDescription,
         coordinator: QBittorrentDataCoordinator,
-        config_entry: ConfigEntry,
+        entity_description: SensorEntityDescription,
     ) -> None:
         """Initialize the qBittorrent sensor."""
         super().__init__(coordinator)
-        self.entity_description = description
-        self._attr_unique_id = f"{config_entry.entry_id}-{description.key}"
-        self._attr_name = f"{config_entry.title} {description.name}"
-        self._attr_available = False
+        self.entity_description = entity_description
+        self._attr_unique_id = (
+            f"{coordinator.config_entry.entry_id}-{entity_description.key}"
+        )
+        self._attr_device_info = DeviceInfo(
+            entry_type=DeviceEntryType.SERVICE,
+            identifiers={(DOMAIN, coordinator.config_entry.entry_id)},
+            manufacturer="QBittorrent",
+        )
+
+
+class QBittorrentSpeedSensor(QBittorrentSensor):
+    """Representation of a QBittorrent speed sensor."""
+
+    _attr_device_class = SensorDeviceClass.DATA_RATE
+    _attr_native_unit_of_measurement = UnitOfDataRate.BYTES_PER_SECOND
+    _attr_suggested_display_precision = 2
+    _attr_suggested_unit_of_measurement = UnitOfDataRate.MEGABYTES_PER_SECOND
 
     @property
-    def native_value(self) -> StateType:
-        """Return value of sensor."""
-        return self.entity_description.value_fn(self.coordinator.data)
+    def native_value(self) -> float:
+        """Return the speed of the sensor."""
+        data = self.coordinator.data
+        return (
+            float(data["server_state"]["dl_info_speed"])
+            if self.entity_description.key == "download"
+            else float(data["server_state"]["up_info_speed"])
+        )
+    
+    
+class QBittorrentStatusSensor(QBittorrentSensor):
+    """Representation of a QBittorrent status sensor."""
+
+    _attr_device_class = SensorDeviceClass.ENUM
+    _attr_options = [STATE_IDLE, STATE_UP_DOWN, STATE_SEEDING, STATE_DOWNLOADING]
+
+    @property
+    def native_value(self) -> str:
+        """Return the value of the status sensor."""
+        download = self.coordinator.data["server_state"]["dl_info_speed"]
+        upload = self.coordinator.data["server_state"]["up_info_speed"]
+        if upload > 0 and download > 0:
+            return STATE_UP_DOWN
+        if upload > 0 and download == 0:
+            return STATE_SEEDING
+        if upload == 0 and download > 0:
+            return STATE_DOWNLOADING
+        return STATE_IDLE
+
+
+class QBittorrentTorrentsSensor(QBittorrentSensor):
+    """Representation of a qBittorrent torrents sensor."""
+
+    @property
+    def native_unit_of_measurement(self) -> str:
+        """Return the unit of measurement of this entity, if any."""
+        return "Torrents"
+
+    @property
+    def extra_state_attributes(self) -> dict[str, Any]:
```

Comment:
Even tho the transmission integration is doing this, we want to avoid this. I think the list of torrents is more suited for a service call now we have responses in them :)

Reply:
I'm curious your thoughts and why this would want to be avoided?

To give my thoughts, the main reason I wanted to update this integration is that I want to be able to view information about my torrents in my dashboard. When I was using Transmission, I did that with [this custom card](https://github.com/amaximus/transmission-card). I have created a version of that card that works with this newly updated qBittorrent integration, I just haven't yet uploaded it to GitHub as that doesn't seem appropriate to do until this is landed.

In my mind, it doesn't make sense to require everyone to setup an automation to call a service every 30 seconds to be able to display information about your torrents in a dashboard's card. I would think that someone adding this integration to their HA instance means their intent is to get clear and current information about what their torrent client is doing.

Reply:
Gotcha. Is there a way to have service calls be called and used by javascript based lovelace cards?

Reply:
@joostlek after some research, it seems that there isn't a way to use the response from service calls in lovelace cards right now. I think that  we should allow this to be stored to attributes (just like the Transmission integration does) for now until the ability to use service call responses in lovelace cards is added. Then this swap to making it a service card response makes sense.

Thoughts?

Reply:
Are we talking about custom cards or default cards? It should be possible in custom cards to use the service response. What are you planning to do with it, use the markdown card?

Reply:
Your last comment is not correct and cards can use the response from service calls (https://github.com/home-assistant/core/pull/98610).
The best example is the service tab in the developer tools.

Reply:
@bramkragten this was to use in a custom card, an spinoff of [this card for transmission](https://github.com/amaximus/transmission-card), so it would be in a custom JS card, not a markdown card.

@edenhaus oh nice! We couldn't find any examples of this and didn't see it in the documentation before. Let me look and see if I can find something using this in a javascript card that i can base it off of.

Reply:
https://discord.com/channels/330944238910963714/351047592588869643/1187036660320456794

Reply:
As a followup to this, I was able to successfully build get_torrents as a service (will go up as a followup pull request) and was able to successfully use that service in a javascript based card (after quite a bit of guess work and finagling). Thanks for helping point me in the right direction! Excited to get these landed and distributed to everyone :)